### PR TITLE
Revamp of service catalog YAML structure.

### PR DESCRIPTION
**Pull request by magnusfeuer**
Added nested namespaces.
Deleted interfaces - They are equivalent to namespaces
Cleaned up struture in general.
Added properties.
Added proposal on how to export VSS into a property structure.
Updated inline documentation.
Updated deployment file to match comfort service

### CODE REVIEW

File: comfort-service.yml [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    # Only one service instance will execute a call. The service discovery
+    # and invocation mechanism is up to the implementation to resolve (possibly
+    # with the help of deployment files).
+    #
+    methods:
+      - name: move
+        description: |
+          Set the desired seat position
+
+        in_arguments: # TODO FYI: Gunnar
```
Comment (gunnarx) [inline comfort-service.yml:181]: Franca uses simply "in" and "out".  I think by context it is clear this deals with arguments to/from the method, so why not align the naming?

File: comfort-service.yml [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    #
+    # Properties are service data elements that are distributed
+    # across one or more subscribers. They are equivalent to
+    # MQTT topics or SOME/IP fields.
+    #
+    # TODO: Define relationship, rules, and semantics
+    #       between properties and VSS signals.
+    #       See vss_integration_proposal.yml for one solution.
+    #
+    properties:
```
Comment (gunnarx) [inline comfort-service.yml:294]: Franca calls them Attribute but I agree Property is a fairly well understood term.
Also, "attribute" has a different and conflicting meaning in VSS, so we should rename either one or the other for sure.

Comment (gunnarx) [reply comfort-service.yml:294]: The more general question is:  Is there a need for properties or should they be described in a "VSS-like model" and referred to instead?

File: vss_integration_proposal.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+#
+# (C) 2021 - Magnus Feuer
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#
+
+#
+# A proposal for how we can translate VSS signals to properties.
```
Comment (gunnarx) [inline vss_integration_proposal.yml:10]: Ah, I see this now.  Still a discussion to have.

File: vss_integration_proposal.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    namespaces:
+      - name: VehicleIdentification
+        description:  Attributes that identify a vehicle
+
+        properties:
+          - name: VIN
+            description: 17-character Vehicle Identification Number (VIN) as defined by ISO 3779
+            type: attribute
+            datatype: string # VSS should have same native types as VSC
+
```
Comment (gunnarx) [inline vss_integration_proposal.yml:45]: This translation of VSS to VSC looks technically correct AFAICT.  If we do this then datatypes should be identical as stated - in fact it appears that **all** metadata fields are and should be identical.

File: comfort-service.yml (hunk 1 above)
Comment (magnusfeuer) [reply comfort-service.yml:181]: Will rename

**Comment by magnusfeuer** [comment]
We probably need to reintroduce interface as a host for methods.
We will look at FrancaIDL to see if interface and namespace can both host datatype definitions, or if they are restricted to one of interface/namespace.

File: comfort-service.yml [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#
+
+
+# Namespace
+#
+#  A n namespace is a collection of datatypes, methods, properties,
```
Comment (erikbosch) [inline comfort-service.yml:12]: A "n" namespace. Typo?

File: comfort-service.yml [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+# + How do we import namespaces, datatypes, etc from other service files?
+#   Importing datatypes used by multiple services make sense, but it
+#   will put a strain on the build pipeline since we now will have to
+#   seach for and assemble a library of interdependent services.
+#
+#
+
+---
+name: comfort
+description: A collection of interfaces pertaining tso cabin comfort.
```
Comment (erikbosch) [inline comfort-service.yml:30]: Minor: "tso". Also "semantiocs" a few lines down

File: comfort-service.yml [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+    methods:
+      - name: move
+        description: |
+          Set the desired seat position
+
+        in_arguments: # TODO FYI: Gunnar
+          - name: seat
+            description: |
+              The desired seat position
+            type: movement.seat_t
```
Comment (erikbosch) [inline comfort-service.yml:185]: What does "movement" refers to here? It is not namespace as that is "seats". Would "type: seat_t" suffice?

**Comment by magnusfeuer** [comment]
Closing this one since it is obsolete after my botched master merge.

**Pull request closed without merging**